lib/core/src/Cardano/Wallet/Primitive/CoinSelection/MA/RoundRobin.hs

```diff
@@ -1219,6 +1439,16 @@ newtype AssetCount a = AssetCount
     { unAssetCount :: a }
     deriving (Eq, Show)
 
+--------------------------------------------------------------------------------
+-- Constants
+--------------------------------------------------------------------------------
+
+-- | The greatest token quantity that can be encoded within an output bundle of
+--   a transaction.
+--
+maxTxOutTokenQuantity :: TokenQuantity
+maxTxOutTokenQuantity = TokenQuantity $ fromIntegral (maxBound :: Word64)
```

Comment:
It's actually half that. The max value that can be encoded must fit in an `int64`, so `9223372036854775807`.